x/leverage/keeper/oracle.go

```diff
@@ -51,7 +55,20 @@ func (k Keeper) TokenDefaultDenomPrice(ctx sdk.Context, baseDenom string) (sdk.D
 		return sdk.ZeroDec(), t.Exponent, types.ErrBlacklisted
 	}
 
-	price, err := k.oracleKeeper.GetExchangeRate(ctx, t.SymbolDenom)
+	var price sdk.Dec
+
+	if historic {
+		// historic price
+		var numStamps uint32
+		price, numStamps, err = k.oracleKeeper.MedianOfHistoricMedians(ctx, t.SymbolDenom, numHistoracleStamps)
+		if err == nil && numStamps == 0 {
+			// if no price medians were available, current price is used as the historic price
+			price, err = k.oracleKeeper.GetExchangeRate(ctx, t.SymbolDenom)
+		}
```

Comment:
Note here:

An `err != nil` return from `MedianOfHistoricMedians` will result in an error and stop any transactions which rely on historic prices from executing.

A `numStamps == 0` return, meaning no historic prices are stored for this denom, will attempt to use current prices instead. If they're available, the function will proceed. If an error occurs instead, it will likely be the usual `unknown denom: UMEE` oracle error for missing prices.

This behavior (assuming the current price is available) is _fail available_ rather than _fail safe_, since historical prices are a guardrail that acts in addition to the main (current prices) borrow restrictions.

Reply:
Outdated - now fails safe